sdl-core/src/main/scala/io/smartdatalake/config/ConfigLoader.scala

```diff
@@ -185,14 +184,17 @@ object ConfigLoader extends SmartDataLakeLogger {
             logger.warn(s"Failed to list directory content of ${nextFile.toString}.", exception)
           case Success(children) =>
             while (children.hasNext) {
-              traversalQueue += children.next().getPath
-              logger.trace(s"Found '${traversalQueue.last.getName}' in directory $nextFile.")
+              val childPath = children.next.getPath
+              if (!childPath.getName.startsWith(".")) { // ignore hidden entries
+                traversalQueue += childPath
+                logger.debug(s"Found '${childPath.getName}' in directory $nextFile.")
+              }
             }
         }
       } else if (fs.isFile(nextFile) && hasPermission(nextFile, FsAction.READ)) {
         val fileExtension = nextFile.getName.split('.').last
-        if (configFileExtensions.contains(fileExtension) && !nextFile.getName.equals("log4j.properties")) {
-          logger.trace(s"'$nextFile' is a configuration file.")
+        if (configFileExtensions.contains(fileExtension) && !nextFile.getName.startsWith("log4j")) {
```

Comment:
Could we use contains instead of startWith ? In that case, log4j would be a reserved word that you're not allowed to use at all. 
Not sure if there are other reserved words we would need to take into account.

Reply:
Good suggestion. I changed it to contains("log4j").